pallets/fragments/src/lib.rs

```diff
@@ -453,34 +664,36 @@ pub mod pallet {
 
 			let price: u128 = sale.price.into();
 
-			let fragment_data = <Classes<T>>::get(fragment_hash).ok_or(Error::<T>::NotFound)?;
+			let fragment_data = <Definitions<T>>::get(fragment_hash).ok_or(Error::<T>::NotFound)?;
 
-			let vault = &Self::get_vault_id(fragment_hash);
+			let vault = &Self::get_vault_id(fragment_hash); // Get the Vault Account ID of `fragment_hash`
 
 			let quantity = match options {
 				FragmentBuyOptions::Quantity(amount) => u64::from(amount),
 				_ => 1u64,
 			};
 
-			let price = price.saturating_mul(quantity as u128);
+			let price = price.saturating_mul(quantity as u128); // `price` = `price` * `quantity`
 
 			// ! Writing if successful
 
 			if let Some(currency) = fragment_data.metadata.currency {
 				<pallet_assets::Pallet<T> as Transfer<T::AccountId>>::transfer(
+					// transfer `price` units of `currency` from `who` to `vault`
 					currency,
 					&who,
 					&vault,
 					price.saturated_into(),
-					true,
+					true, // investigate ???
```

Comment:
TODO I guess? Feel free to ask

Reply:
`keep_alive` is related to this one? [https://wiki.polkadot.network/docs/glossary#keep-alive-check ](https://wiki.polkadot.network/docs/glossary#keep-alive-check )

> The keep-alive check is used to indicate whether or not a transfer can allow the sending account to be reduced to less than the existential deposit, causing it to be reaped.

Yeah, I think it is the same check that can also be found in the Balance pallet [here](https://github.com/paritytech/substrate/blob/8310936bd25519cee81abb01d2b164805f01bc25/frame/balances/src/lib.rs#L270) in the `transfer` function:

> `transfer_keep_alive` works the same way as `transfer`, but has an additional check that the transfer will not kill the origin account.

Reply:
This is useful. Thanks @alebaffa